hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -228,10 +289,8 @@ private OmKeyInfo createDirectoryKeyInfo(OzoneManager ozoneManager,
         .setReplicationType(HddsProtos.ReplicationType.RATIS)
         .setReplicationFactor(HddsProtos.ReplicationFactor.ONE)
         .setFileEncryptionInfo(encryptionInfo.orNull())
-        .setAcls(OzoneAclUtil.fromProtobuf(keyArgs.getAclsList()))
-        .setObjectID(transactionLogIndex)
-        .setUpdateID(transactionLogIndex)
-        .build();
+        .setObjectID(objectId)
+        .setUpdateID(0);
```

Comment:
Let's not set the updateID here as it will be set later.

Reply:
sure, will make the change. any specific reason why it might not be a good idea?

Reply:
No strong reason. Just to avoid setting it twice.

But this leads me to another thought. Instead of having a separate setObjectID and setUpdateID methods, we could have 1 method which updates both. This would ensure that we never miss setting the updateID when setting objectID. Just a thought.. unrelated to this Jira.